content/200-concepts/100-components/02-prisma-client/260-client-extensions/140-shared-extensions.mdx

```diff
@@ -0,0 +1,121 @@
+---
+title: 'Shared Prisma Client extensions'
+metaTitle: 'Shared Prisma Client extensions (Preview)'
+metaDescription: 'Share extensions or import shared extensions into your Prisma project'
+tocDepth: 3
+---
+
+<TopBlock>
+
+<Admonition type="warning">
+
+This page is in draft.
+
+</Admonition>
+
+You can share your [Prisma Client extensions](/concepts/components/prisma-client/client-extensions/) with other users, either as packages or as code snippets, and import extensions that other users create into your Prisma projects.
+
+</TopBlock>
+
+## Share Prisma Client extensions
+
+When you want to create extensions that will work for other users, and not just for your specific schema, use the `$all` prefix, such as [`$allModels`](/concepts/components/prisma-client/client-extensions/model#add-a-custom-method-to-all-models-in-your-schema). Extensions that use `$all` apply generally, not just to a named component of a specific schema.
+
+For example, a generic extension might take the following form:
+
+```ts
+const xprisma = prisma.$extends({
+  model: {
+    $allModels: {
+      // new method
+      findOrCreate(...) { } // code for the new method goes inside the brackets
+    }
+  }
+})
+
+```
+
+### Package an extension
+
+For extensions that you package and share, you cannot predict which `prisma` instance that the extension user will use. Therefore, you cannot use `prisma.$extends`, which applies to a specific instance. Instead, use `Prisma.defineExtension`. `Prisma.defineExtension` declares the extension but does not apply the extension to a specific instance. In your code editor, it gives you the same autocomplete for the extension as you get for `$extends`.
+
+When someone wants to use your extension in their project, they can apply the extension to an instance with `prisma.$extends`.
```

Comment:
```suggestion
When someone wants to use your extension in their project, they can apply the extension to a Prisma Client instance with `prisma.$extends`.
```

Reply:
I've brought in some of Pierre's suggestions for the introduction here, so this suggestion is out of date, but I'll make the equivalent change in the new intro.